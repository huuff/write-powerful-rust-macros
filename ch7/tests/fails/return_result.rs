use panic_to_result_macro::panic_to_result;

struct Person {
    age: u8,
    name: String,
}

#[panic_to_result]
fn create_person_with_result(age: u8, name: String) -> Result<Person, String> {
    if age > 30 {
        panic!("I hope I die before I get old")
    }

    Ok(Person { age, name })
}

fn main() {}
