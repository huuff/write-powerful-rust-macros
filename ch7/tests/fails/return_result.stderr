error: this macro can only be applied to a function that doesn't return a result. Signature: Result < Person, String >
 --> tests/fails/return_result.rs:9:1
  |
9 | fn create_person_with_result(age: u8, name: String) -> Result<Person, String> {
  | ^^
